src/App.jsx

```diff
@@ -14,33 +18,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const goodsWithIds = goodsFromServer.map(good => {
+  return {
+    name: good,
+    id: uuid(),
+  };
+});
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function getPreparedGoods(goods, { sortField }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.name.localeCompare(good2.name);
+        case SORT_FIELD_LENGTH:
+          return good1.name.length - good2.name.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  let visibleGoods = getPreparedGoods(goodsWithIds, { sortField });
+
+  if (reversed) {
+    visibleGoods = visibleGoods.toReversed();
```

Comment:
The method `toReversed()` is not a standard JavaScript array method. You should use the `reverse()` method instead, but be cautious as `reverse()` mutates the original array. To avoid mutation, you can use `visibleGoods.slice().reverse()` to create a reversed copy of the array.